portability/file.cc

```diff
@@ -72,67 +72,98 @@ void toku_fs_get_write_info(time_t *enospc_last_time, uint64_t *enospc_current,
-	snprintf(err_msg, sizeof(err_msg), "Write of [%" PRIu64 "] bytes to fd=[%d] interrupted.  Retrying.", (uint64_t)len, fd);
-	perror(err_msg);
-	fflush(stderr);
-	try_again = 1;
-	break;
-    }
-    case ENOSPC: {
-        if (toku_assert_on_write_enospc) {
-            char err_msg[sizeof("Failed write of [] bytes to fd=[].") + 20+10]; //64 bit is 20 chars, 32 bit is 10 chars
-            snprintf(err_msg, sizeof(err_msg), "Failed write of [%" PRIu64 "] bytes to fd=[%d].", (uint64_t)len, fd);
-            perror(err_msg);
-            fflush(stderr);
-            int out_of_disk_space = 1;
-            assert(!out_of_disk_space); //Give an error message that might be useful if this is the only one that survives.
-        } else {
-            toku_sync_fetch_and_add(&toku_write_enospc_total, 1);
-            toku_sync_fetch_and_add(&toku_write_enospc_current, 1);
-
-            time_t tnow = time(0);
-            toku_write_enospc_last_time = tnow;
-            if (toku_write_enospc_last_report == 0 || tnow - toku_write_enospc_last_report >= 60) {
-                toku_write_enospc_last_report = tnow;
-
-                const int tstr_length = 26;
-                char tstr[tstr_length];
-                time_t t = time(0);
-                ctime_r(&t, tstr);
-
-                const int MY_MAX_PATH = 256;
-                char fname[MY_MAX_PATH], symname[MY_MAX_PATH+1];
-                sprintf(fname, "/proc/%d/fd/%d", getpid(), fd);
-                ssize_t n = readlink(fname, symname, MY_MAX_PATH);
-
-                if ((int)n == -1)
-                    fprintf(stderr, "%.24s PerconaFT No space when writing %" PRIu64 " bytes to fd=%d ", tstr, (uint64_t) len, fd);
-                else {
-		    tstr[n] = 0; // readlink doesn't append a NUL to the end of the buffer.
-                    fprintf(stderr, "%.24s PerconaFT No space when writing %" PRIu64 " bytes to %*s ", tstr, (uint64_t) len, (int) n, symname);
-		}
-                fprintf(stderr, "retry in %d second%s\n", toku_write_enospc_sleep, toku_write_enospc_sleep > 1 ? "s" : "");
-                fflush(stderr);
-            }
-            sleep(toku_write_enospc_sleep);
+        case EINTR: {  // The call was interrupted by a signal before any data
+                       // was written; see signal(7).
+            report_io_problem(fd,
+                              "Write of [%" PRIu64
+                              "] bytes interrupted.  Retrying.",
+                              (uint64_t)len);
             try_again = 1;
-            toku_sync_fetch_and_sub(&toku_write_enospc_current, 1);
             break;
         }
-    }
-    default:
-	break;
+        case ENOSPC: {
+            if (toku_assert_on_write_enospc) {
+                report_io_problem(
+                    fd, "Failed write of [%" PRIu64 "] bytes.", (uint64_t)len);
+                int out_of_disk_space = 1;
+                assert(!out_of_disk_space);  // Give an error message that might
```

Comment:
Huh. bool at least?

Reply:
This was original FT code and I saw no reason to change it, funky as it is.

Reply:
oh, and static constexpr yada yada *if* you decide to touch it